include/seqan3/io/detail/take_until_view.hpp

```diff
@@ -0,0 +1,684 @@
+ * | std::ranges::input_range         | *required*                            | *preserved*                                        |
+ * | std::ranges::forward_range       |                                       | <i>preserved</i>¹                                  |
+ * | std::ranges::bidirectional_range |                                       | <i>preserved</i>¹                                  |
+ * | std::ranges::random_access_range |                                       | <i>preserved</i>¹                                  |
+ * | std::ranges::contiguous_range    |                                       | <i>preserved</i>¹                                  |
+ * |                                  |                                       |                                                    |
+ * | std::ranges::viewable_range      | *required*                            | *guaranteed*                                       |
+ * | std::ranges::view                |                                       | *guaranteed*                                       |
+ * | std::ranges::sized_range         |                                       | *lost*                                             |
+ * | std::ranges::common_range        |                                       | *lost*                                             |
+ * | std::ranges::output_range        |                                       | *preserved*                                        |
+ * | seqan3::const_iterable_range     |                                       | <i>preserved</i>¹                                  |
+ * |                                  |                                       |                                                    |
+ * | std::ranges::range_reference_t   |                                       | std::ranges::range_reference_t<urng_t>             |
+ *
+ * See the \link views views submodule documentation \endlink for detailed descriptions of the view properties.
+ *
+ * ¹ The marked properties are only *preserved* if the specified functor models
+ * `std::regular_invocable<fun_t, std::ranges::range_reference_t<urng_t>`, i.e. applying the functor doesn't change the functor.
+ * If the functor only models `std::invocable` and not `std::regular_invocable` these concepts are *lost*.
+ *
+ * Throwing: `seqan3::detail::take_until_or_throw` and `seqan3::detail::take_until_or_throw_and_consume` throw an exception
+ * if the end of the underlying range is reached before their own termination criterium is met. This is useful
+ * if you want a "strict" evaluation of the functor.
+ *
+ * Consuming: `seqan3::detail::take_until_and_consume` and `seqan3::detail::take_until_or_throw_and_consume` behave the
+ * same as their non-consuming counter-parts if the underlying range models at least `std::range::forward_range`.
+ * If, however, the underlying range is a pure `std::range::input_range`, the view will keep moving the underlying
```

Comment:
As it is the `ranges` library it must be `std::ranges` instead of `std::range`, right? (https://en.cppreference.com/w/cpp/ranges)
```suggestion
 * same as their non-consuming counter-parts if the underlying range models at least `std::ranges::forward_range`.
 * If, however, the underlying range is a pure `std::ranges::input_range`, the view will keep moving the underlying
```

Reply:
Oh no! yes you right! thanks!